### PR TITLE
test(openchallenges): improve organization service service unit test coverage

### PR DESCRIPTION
**Pull request by mdsage1**
## Description

Sonar unit organization service test coverage has been increased to 52%, from 19%, for service package by writing unit tests for a single class, OrganizationService.

## Related Issue

#2389
#2063 
#1528 
#2348 
#2388  

Fixes #(issue)

## Changelog

See `files Changed` section
- Add 2 unit tests for getOrganization and other methods

### CODE REVIEW

**Comment by mdsage1** [comment]
Sonar suggested I remove all public modifiers so I removed them.

**Comment by mdsage1** [comment]
PR is ready for review.  2 Unit tests have been added for OrganizationService. Line 68 of the code has a public modifier which Sonar suggests I remove but that causes a formatting issue. @tschaffter Let me know which error/issue tracker takes priority here. I've prioritized the springBoot built in formatting.

**Comment by mdsage1** [comment]
@tschaffter please review my PR

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java [hunk 1]
```diff
@@ -1,81 +1,91 @@
...
+
+    OrganizationService organizationService = new OrganizationService(organizationRepository);
+
+    // get the organization using the identifier
+    OrganizationDto response = organizationService.getOrganization(identifier);
+
+    /* Test that the organization is in the repo, is not null,
+    that the id can be pulled, findByIdOrLogin method is called,
+    and no interactions with the organizationMapper
+    (verifies interactions w/Mock object) */
```
Comment (tschaffter) [inline apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java:59]: Use 80-char lines for code and comment

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java [hunk 2]
```diff
@@ -1,81 +1,91 @@
...
+
+    // Test that calling getOrganization throws an error
+    assertThrows(
+        OrganizationNotFoundException.class,
+        () -> {
+          organizationService.getOrganization(invalidIdentifier);
+        });
+    /* Test thatfindByIdOrLogin method is called,
+    and no interactions with the organizationMapper
+    (verifies interactions w/Mock object)*/
```
Comment (tschaffter) [inline apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java:86]: Same comment as above

**Comment by tschaffter** [comment]
> PR is ready for review. 2 Unit tests have been added for OrganizationService. Line 68 of the code has a public modifier which Sonar suggests I remove but that causes a formatting issue. @tschaffter Let me know which error/issue tracker takes priority here. I've prioritized the springBoot built in formatting.

Is this issue still relevant? It looks like you resolved the formating issue (with `nx run openchallenges-organization-service:format`?).

You are adding two unit tests to the organization service in this PR but the Sonar app does not report an increase in coverage. Do you know why? Do you see an increase in coverage using JaCoCo?

**Comment by mdsage1** [comment]
> > PR is ready for review. 2 Unit tests have been added for OrganizationService. Line 68 of the code has a public modifier which Sonar suggests I remove but that causes a formatting issue. @tschaffter Let me know which error/issue tracker takes priority here. I've prioritized the springBoot built in formatting.
> 
> Is this issue still relevant? It looks like you resolved the formating issue (with `nx run openchallenges-organization-service:format`?).
> 
> You are adding two unit tests to the organization service in this PR but the Sonar app does not report an increase in coverage. Do you know why? Do you see an increase in coverage using JaCoCo?

@tschaffter Yes I have documented the increase in coverage seen in Jacoco, from 19% to 52%, in the Issue #1528 which is what this PR is addressing. I left the Sonar only information that is auto generated in this PR since I thought it might be overkill to add both. I did resolve the formatting issue after making the comment about it.

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java (hunk 1 above)
Comment (mdsage1) [reply apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java]: I've made the changes to the characters on the comment line.

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java (hunk 2 above)
Comment (mdsage1) [reply apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java]: Updated the character count in this line of code.

**Comment by mdsage1** [comment]
> > > PR is ready for review. 2 Unit tests have been added for OrganizationService. Line 68 of the code has a public modifier which Sonar suggests I remove but that causes a formatting issue. @tschaffter Let me know which error/issue tracker takes priority here. I've prioritized the springBoot built in formatting.
> > 
> > 
> > Is this issue still relevant? It looks like you resolved the formating issue (with `nx run openchallenges-organization-service:format`?).
> > You are adding two unit tests to the organization service in this PR but the Sonar app does not report an increase in coverage. Do you know why? Do you see an increase in coverage using JaCoCo?
> 
> @tschaffter Yes I have documented the increase in coverage seen in Jacoco, from 19% to 52%, in the Issue #1528 which is what this PR is addressing. I left the Sonar only information that is auto generated in this PR since I thought it might be overkill to add both. I did resolve the formatting issue after making the comment about it.

@tschaffter I've made the requested changes to the comment lines character counts.

**Comment by mdsage1** [comment]
PR marked as draft after noticing that there were new changes upstream that needed to be synced to current repo branch. This triggered a number of test failures related to schematic_api/...
@tschaffter I'm unsure what has occurred here as none of this is connected to the unit tests I've implemented. Do you have any idea what's happened? It looks like PR #2532 has no evidence of CI/CD checks but has a dpull check listed. I can't decipher if the issue came from this PR but it does indicate there was a change to the latest CSV dump files. Here are the errors:

> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetNodeDependencyArray::test_return_display_names - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetNodeDependencyArray::test_return_ordered_by_schema - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetNodeDependencyArray::test_success - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetNodeDependencyPage::test_success - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetNodeDependencyPage::test_return_ordered_by_schema - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetNodeDependencyPage::test_return_display_names - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetNodeIsRequired::test_success - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetNodeValidationRules::test_success - AssertionError: assert ['detail', 'status', 'title'] == ['validation_rules']
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetConnectedNodePairPage::test_success - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetSchemaAttributes::test_success - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetConnectedNodePairArray::test_success - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetPropertyLabel::test_success - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetComponent::test_parameters - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetComponent::test_success - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_endpoints.py::TestGetNodeProperties::test_success - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_schema_controller_impl.py::TestGetNodeValidationRuleArray::test_success - assert 500 == 200
> FAILED schematic_api/test/test_schema_controller_impl.py::TestGetComponent::test_success - assert 500 == 200
> FAILED schematic_api/test/test_schema_controller_impl.py::TestGetConnectedNodePairArray::test_success - assert 500 == 200
> FAILED schematic_api/test/test_schema_controller_impl.py::TestGetNodeDependencyPage::test_success - assert 500 == 200
> FAILED schematic_api/test/test_schema_controller_impl.py::TestGetNodeIsRequired::test_success - assert 500 == 200
> FAILED schematic_api/test/test_schema_controller_impl.py::TestGetNodeProperties::test_success - assert 500 == 200
> FAILED schematic_api/test/test_schema_controller_impl.py::TestGetSchemaAttributes::test_success - assert 500 == 200
> FAILED schematic_api/test/test_schema_controller_impl.py::TestGetConnectedNodePairPage::test_success - assert 500 == 200
> FAILED schematic_api/test/test_schema_controller_impl.py::TestGetNodeDependencyArray::test_success - assert 500 == 200
> FAILED schematic_api/test/test_schema_controller_impl.py::TestGetPropertyLabel::test_success - assert 500 == 200
> FAILED schematic_api/test/test_manifest_validation_controller_impl.py::TestValidateManifestJson::test_success_no_errors - assert 500 == 200
> FAILED schematic_api/test/test_manifest_validation_controller_impl.py::TestValidateManifestJson::test_success_one_error - assert 500 == 200
> FAILED schematic_api/test/test_manifest_validation_controller_impl.py::TestValidateManifestCsv::test_success_no_errors - assert 500 == 200
> FAILED schematic_api/test/test_manifest_validation_controller_impl.py::TestValidateManifestCsv::test_success_with_one_error - assert 500 == 200
> FAILED schematic_api/test/test_tangled_tree_impl.py::TestGetTangledTreeLayers::test_success - assert 500 == 200
> FAILED schematic_api/test/test_tangled_tree_impl.py::TestGetTangledTreeText::test_success - assert 500 == 200
> FAILED schematic_api/test/test_manifest_validation_endpoints.py::TestValidateManifestJson::test_success - AssertionError: 500 != 200 : Response body is : {
> FAILED schematic_api/test/test_manifest_validation_endpoints.py::TestValidateManifestCsv::test_success - AssertionError: 500 != 200 : Response body is : {
> === 33 failed, 185 passed, 18 deselected, 4579 warnings in 74.29s (0:01:14) ====
> Warning: run-commands command "poetry run pytest -m 'not secrets'" exited with non-zero status code
>  
> 
>  >  NX   Running target test for 17 projects and 4 tasks they depend on failed
> 
>    Failed tasks:
>    
>    - schematic-api:test
> 
> Error: Process completed with exit code 1.

**Comment by tschaffter** [comment]
Hey @andrewelamb, Maria bumped into the above Schematic test issues after updating her feature branch with `main`. Do these test failures seem familiar?

**Comment by andrewelamb** [comment]
@tschaffter There was a change to schematic that intorduced soem breaking changes the the API in the monorepo, involving both claled methods and the data model used for testing. These were fixed by [this PR.](https://github.com/Sage-Bionetworks/sage-monorepo/pull/2515). So if [this commit](https://github.com/Sage-Bionetworks/sage-monorepo/pull/2515/commits/61a122bf3eb73637b5a9c3b2316a54a26dd49da3) isn't in the branch the schematic API tests will fail.

**Comment by andrewelamb** [comment]
@tschaffter Just curious, why are the schematic API tests beign run for this PR?

**Comment by tschaffter** [comment]
> @tschaffter Just curious, why are the schematic API tests beign run for this PR?

That's an excellent question. I will document here my findings here as I explore this issue.

## Troubleshooting

[Here](https://github.com/Sage-Bionetworks/sage-monorepo/actions/runs/8070788847/job/22048945760) are the logs of the CI workflow that run for the last commit pushed by Maria to this PR. One of the first job of the workflow shows the "Base SHA" and "Head SHA" that `nx` uses to identify the list of affected projects for which tasks like `lint`, `build`, `test`, etc. will be run by the CI workflow.

Base SHA
ff72e18bec3b1b3a64b0566aac914b9ddb07caa7

Head SHA
75645b67a4fc602aef94f363006ac5f7aef17571

**Comment by andrewelamb** [comment]
@tschaffter After further investigation in the current version of main, the schemati api tests are faling as well, this also has to do with the breaking changes made by the most recent version of schematic. This will be fixed by [this PR.](https://github.com/Sage-Bionetworks/sage-monorepo/pull/2533)

**Comment by tschaffter** [comment]
> @tschaffter Just curious, why are the schematic API tests beign run for this PR?

@andrewelamb I can now answer your question (see [details](https://github.com/Sage-Bionetworks/sage-monorepo/pull/2507#issuecomment-1967781495)). In a nutshell, Nx identifies the **affected** projects using two SHA: the Head and Base SHAs. The Head SHA refers to the commit ID of the last commit pushed to a feature branch/PR. Determining the Base SHA is a bit more tricky:

- Expected: The Base SHA should refer to the last successful commit pushed to the `main` branch of the official repo (this repo).

- Actual: The Base SHA refers to the last successful commit pushed to the `main` branch of the fork that opened the PR. The key lesson here is the reference to the `main` branch of the fork, not of the official repo. Hence, it is important to keep your fork's `main` branch of to date before pushing new commits.

I will update the CI workflow so that the Base SHA refers to the `main` branch of the monorepo.

**Comment by mdsage1** [comment]
@tschaffter Can you review this pr please?

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java [hunk 3]
```diff
@@ -1,81 +1,89 @@
...
+@ExtendWith(MockitoExtension.class)
+class OrganizationServiceTest {
+
+  @Mock private OrganizationRepository organizationRepository;
+
+  @Mock private OrganizationMapper organizationMapper;
+
+  @BeforeEach
+  void setup() {
+    OrganizationService organizationService = new OrganizationService(organizationRepository);
```
Comment (tschaffter) [inline apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java]: `organizationService` is never used

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java [hunk 4]
```diff
@@ -1,81 +1,89 @@
...
-//   }
-
-// }
+package org.sagebionetworks.openchallenges.organization.service.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
```
Comment (tschaffter) [inline apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java]: Do not import the entire package.

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java (hunk 3 above)
Comment (tschaffter) [reply apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java]: I have fixed these issues in the commits below

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java (hunk 4 above)
Comment (tschaffter) [reply apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/service/OrganizationServiceTest.java]: I have fixed these issues in the commits below

**Pull request merged**